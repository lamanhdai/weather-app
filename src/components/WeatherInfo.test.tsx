import React from 'react';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

import WeatherInfo from './WeatherInfo';

describe('WeatherInfo', () => {
  it('render weather element', () => {
    const info = {
      id: '',
      weather_state_name: '',
      weather_state_abbr: '',
      wind_direction_compass: '',
      applicable_date: '',
      min_temp: 0,
      max_temp: 0,
      the_temp: 0,
      wind_speed: 0,
      wind_direction: 0,
      air_pressure: 0,
      humidity: 0,
      visibility: 0,
      predictability: ''
    }
    const WeatherElement = renderer
      .create(<WeatherInfo info={info} />)
      .toJSON();

    expect(WeatherElement).toMatchSnapshot();
  });
})