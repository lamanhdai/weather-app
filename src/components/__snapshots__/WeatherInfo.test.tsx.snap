// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo render weather element 1`] = `
<div
  className="sc-dlfnuX dcvreJ mt-4 h-100 d-flex flex-column"
>
  <h5>
    Invalid date
  </h5>
  <div
    className="mt-auto text-center"
  >
    <div>
      <span
        className="sc-bdfBQB cgoigE"
      />
    </div>
    <span>
      
    </span>
  </div>
  <div
    className="mt-4"
  >
    <strong>
      Temp
    </strong>
    <p>
      Max: 
      0
      °C
    </p>
    <p>
      Min: 
      0
      °C
    </p>
  </div>
  <div>
    <span
      className="sc-gsTEea eFzlKo"
    />
    0
    mph
  </div>
</div>
`;
